agent/structs/config_entry_discoverychain.go

```diff
@@ -943,6 +1022,202 @@ type ServiceResolverFailover struct {
 	Datacenters []string `json:",omitempty"`
 }
 
+// LoadBalancer determines the load balancing policy and configuration for services
+// issuing requests to this upstream service.
+type LoadBalancer struct {
+	// EnvoyLBConfig contains Envoy-specific load balancing configuration for this upstream
+	EnvoyLBConfig *EnvoyLBConfig `json:",omitempty" alias:"envoy_lb_config"`
```

Comment:
Since the enclosing struct is `LoadBalancer` maybe the envoy config doesn't need to repeat the LB infix in the names (you still need it for disambiguation as a Go type in this package) and could just be

```
EnvoyConfig *EnvoyLBConfig `json:",omitempty" alias:"envoy_config"`
```